Libraries/SDHC/ff13/Source/diskio.c

```diff
@@ -101,17 +165,57 @@ DRESULT disk_read (
 	UINT count		/* Number of sectors to read */
 )
 {
-    DRESULT status;
+	if(!flash){
+    	DRESULT status;
 
-    if (MXC_SDHC_Lib_Read(buff, sector, count, MXC_SDHC_LIB_SINGLE_DATA) != E_NO_ERROR) {
-		status = RES_ERROR;
-    } else {
-		status = RES_OK;
+    	if (MXC_SDHC_Lib_Read(buff, sector, count, MXC_SDHC_LIB_SINGLE_DATA) != E_NO_ERROR) {
+			status = RES_ERROR;
+    	} else {
+			status = RES_OK;
+    	}
+
+    	return status;
+	}
+#ifdef FLASH
+	else{
+		    DRESULT status;
+
+    int index = 0;
+
+    status = RES_OK;
+
+#if FF_MAX_SS == 4096
```

Comment:
I am a little confused on the driver differences between `FF_MAX_SS = 4096` and `FF_MAX_SS = 512`.  

To improve the readability of the code I would suggest adding a comment explaining it, and you can also consolidate inside of the for loop:

```C
for(index = 0; index < count; index++){
#if FF_MAX_SS == 4096
    // ...
#endif
#if FF_MAX_SS == 512
    // ...
#endif
}
```

Reply:
I took the base code from the MAXREFDES104 project, and FF_MAX_SS = 4096 part was added to their code. I will remove all FF_MAX_SS = 4096 parts.